### ISSUE TITLE
Punt Yards bug

### ISSUE THREAD

**Issue by TheMathNinja** [nflverse/nflfastR#53]
I'm noticing that kick_distance, which generally fills in for Punt Yards on punts, is NA on Touchback plays. But this should be showing the distance of the punt still, not NA.

This might apply to any time a punt was kicked into the end zone, as for these punts: 
2424 | 2019_12_DET_WAS
5456 | 2018_11_KC_LA

I'm not sure if a punt goes from the 50 to the -2 yard line if that's a 50 or 52 yard punt in the NFL's books, but I think kick_distance should be consistent with whatever NFL counts as "punt yards" on the play.

**Comment by TheMathNinja** [comment]
I also see that kick_distance is NA on blocked punts, but I'd like to advocate for kick_distance to be 0 on blocked punts, since punt_attempt == 1, there actually was a "kick" on the play.

**Comment by TheMathNinja** [comment]
Found official NFl language on measuring kick_distance for punts here: 
"Punts shall be measured from the line of scrimmage to the point at which the impetus of the punt
ends, or the ball goes out of bounds, or is downed. If a punt goes into the end zone, measure its
length to the goal line, unless the ball is fielded and returned out of the end zone by the receiving
team (in such a case, measure the punt to the spot of possession in the end zone by the receiver)."

**Comment by TheMathNinja** [comment]
Found another example of kick_distance == NA on a punt. This play (3267 | 2015_04_DET_SEA) was a muffed return which is also showing NA kick_distance for some reason. I'm thinking as long as punt_attempt == 1, kick_distance should never be NA.

**Comment by guga31bb** [comment]
These were omissions on our part as the stat IDs were present- thank you!

**Comment by mrcaseb** [comment]
The missing kick distance for punts to the end zone has been fixed using stat id 31. However, this stat id is only recorded if the returner exits the end zone. Not for touchbacks.

> SuperStat records this stat when the punt is received in the end-zone, and then run out of the end-zone. If the play ends in the end-zone for a touchback, the stat is not recorded. This stat is used exclusively of the PU, PU_TB, and PU_BK stats. THIS STAT DOESN'T MAKE SENSE TO ME BECAUSE IT'S USUALLY CONSIDERED A GOOD PUNT IF THE BALL IS KICKED HIGH ENOUGH TO PREVENT A RETURN. THIS STAT IS UNKNOWN TO ELIAS & THE NFL.

**Comment by mrcaseb** [comment]
We could probably use `yardline_100` for punts with NA kickdistance

**Comment by john-b-edwards** [comment]
This was brought up in the discord tonight, throwing my hat into the ring of something like

```r
kick_distance = if_else(punt_attempt == 1 & punt_blocked == 0, readr::parse_number(stringr::str_extract(desc, "punts ([0-9])+ yards")), kick_distance)
```
?

dislike using regex but if it's not recorded for the specific stat ID's I'm unsure what to use otherwise

**Comment by mrcaseb** [comment]
For future reference: final fix of this issue happened in #422 by using the corresponding stat IDs
